pkg/node/provisioners/drive_provisioner.go

```diff
@@ -224,10 +226,23 @@ func (d *DriveProvisioner) GetVolumePath(vol *api.Volume) (string, error) {
 	}
 	volumeUUID, _ = util.GetVolumeUUID(volumeUUID)
 
-	partNum, err := d.partOps.SearchPartName(device, volumeUUID)
-	if err != nil {
-		// on device disconnect or node reboot device name might change and we need to re-sync drive info
-		return "", fmt.Errorf("unable to find part name for device %s by uuid %s: %w", device, volumeUUID, err)
+	// For volumes that are in Published status, ignore partition re-read and
+	// get the partition number directly.
+	var partNum string
+	if partNum, err = d.partOps.SearchPartName(device, volumeUUID); err != nil {
+		// blockdev command run return exec.ExitError, exit error code 1 means
+		// the blockdev hit "device or resource busy" error.
+		if exitError, ok := err.(*exec.ExitError); ok &&
+			strings.Contains(string(exitError.Stderr), "Device or resource busy") &&
+			vol.CSIStatus == apiV1.Published {
```

Comment:
Could we just check that device is really mounted when we see `"Device or resource busy"` error and not operate with volume statuses? In that case we can make error check a part of `SearchPartName` function

Reply:
thanks @safronovD, @yurkov-anton  for the suggestions. Yes,  we can check the device mounted or not to avoid using volume status in SearchPartName.  
here is my questions,  for raw block mode  bind mount and file system mount,  do we have API that can tell if it is mounted or not ?  Please suggest, thank you.

Reply:
Hi @safronovD, @yurkov-anton,  updated code based on your latest comments.  Could you help review again?

Reply:
Sorry for delay. We already have the same method to check mount status - https://github.com/dell/csi-baremetal/blob/eccff10da812669bbab4c529c45e66be8c813972/pkg/base/linuxutils/fs/fs_helper.go#L235

Reply:
Do you resolve problem with the raw block mode?

Reply:
Hi @safronovD,  I see we have this similar function in fs_helper.go,  but we can't use it easily in SeachPartName fuction(partition_operations.go),  since It is implemented for WrapFSImpl.  
another reason is it can't meet our requirement,   since what we used there, the input parameters are something like "/dev/sdh",  we need to remove the /dev prefix to find it in /proc/self/mountinfo. 

for your question on raw block mode,  I already checked,  the /proc/slef/mountinfo can find similar entry in it.  so this method is enough.

Reply:
Hi @safronovD,  @yurkov-anton,  may you help take a look at my above comments ?

Reply:
Hi @safronovD,  @yurkov-anton,  sorry to bother you again,  could you please take a look and give any comments or approve the PR ?  Thanks.

Reply:
1. Need to add WrapFSImpl to [PartitionOperations](https://github.com/libzhang/csi-baremetal/blob/620565c880d9a0485f6283f0c2008f3be45040f8/pkg/node/provisioners/utilwrappers/partition_operations.go#L43)
2. I'm not sure need to remove the `/dev` part
```
cat /proc/self/mountinfo | grep /dev/sda
132 99 8:1 / /boot rw,relatime shared:72 - ext4 /dev/sda1 rw,seclabel
```

Reply:
Hi @yurkov-anton, sorry for missing your last comments. Here is why I propose to remove the /dev part,  since I checked on a hdd cluster,  it shows below results,  there is no /dev prefix here,  so we need to cover all scenario here,  remove the prefix is more robust.
```
# cat /proc/self/mountinfo | grep sdi
3815 2218 0:6 /sdi1 /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-c9f66ea3-69ca-4516-a84a-5f6e53ea43f9/globalmount/dev rw,nosuid,relatime shared:626 - devtmpfs udev rw,size=32771740k,nr_inodes=8192935,mode=755
3816 2176 0:6 /sdi1 /hostroot/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-c9f66ea3-69ca-4516-a84a-5f6e53ea43f9/globalmount/dev rw,nosuid,relatime shared:626 - devtmpfs udev rw,size=32771740k,nr_inodes=8192935,mode=755
#
```

Reply:
and another nvme cluster,  it shows below,
```
admin@provo-rust:~> cat /proc/self/mountinfo  | grep nvme1n1
8941 127 0:5 /nvme1n1 /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/staging/pvc-644c98ad-9013-4db7-8392-9ad057f6fddb/dev rw,nosuid,noexec shared:1532 - devtmpfs devtmpfs rw,size=4096k,nr_inodes=1048576,mode=755
8956 127 0:5 /nvme1n1 /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/publish/pvc-644c98ad-9013-4db7-8392-9ad057f6fddb/a786470e-098d-4dfa-957b-94742b0c6925 rw,nosuid,noexec shared:1532 - devtmpfs devtmpfs rw,size=4096k,nr_inodes=1048576,mode=755
9184 127 0:5 /nvme1n1 /var/lib/kubelet/plugins/kubernetes.io/csi/volumeDevices/pvc-644c98ad-9013-4db7-8392-9ad057f6fddb/dev/a786470e-098d-4dfa-957b-94742b0c6925 rw shared:1532 - devtmpfs devtmpfs rw,size=4096k,nr_inodes=1048576,mode=755

```

Reply:
Hi @libzhang, I got it. Then I think you need remove `/dev` part before call `isMounted`. If both of check check are exist, there will be a conflict. Check with name `isMounted` must check path in mount proc. This check doesn't know about what this path is mean, path to dir or to drive. If you want you can create check for drive, which will be contained removing `/dev` part and `isMounted` call.

Reply:
Hi @yurkov-anton,  sorry,  I did not fully understand your suggestions,  do you mean below ?  
1) remove "/dev" before call isMounted
2) using the WrapFsImpl   implemented IsMounted function?

Reply:
@yurkov-anton,  I updated diff based on above my understanding.

Reply:
Yep, it's ok)